rdr_service/dao/dv_order_dao.py

```diff
@@ -138,6 +137,13 @@ def from_client_json(
             if not existing_obj:
                 raise NotFound("existing order record not found")
 
+            # handling of biobankStatus from Mayolink API
+            try:
+                existing_obj.biobankStatus = resource_json['biobankStatus']
```

Comment:
resource json here would be the payload from the request which will not contain `biobankStatus`.

Reply:
Shouldn't `biobankStatus` exist in a PUT request here, set in `to_client_json`? The `model` parameter would be the response from MayoLink, and `for_update` is `True`.
Otherwise you are right, it would not exist, hence the catching on `KeyError` below.

Reply:
If biobank status is the status of the biobank order from the mayolink api then PTSC should not be updating it.